xtask/src/commands/release.rs

```diff
@@ -87,6 +87,19 @@ impl Prepare {
 
         if let Version::Nightly = &self.version {
             println!("Skipping various steps because this is a nightly build.");
+            // Only update helm charts on specific arch/os/env
+            if cfg!(target_arch = "x86_64") && cfg!(target_os = "linux") && cfg!(target_env = "gnu")
+            {
+                // Update the image repository to use the nightly location
+                replace_in_file!(
+                    "./helm/chart/router/values.yaml",
+                    "^  repository: ghcr.io/apollographql/router$",
+                    format!("  repository: ghcr.io/apollographql/nightly/router")
+                );
+
+                // Update the version string for nightly builds
+                self.update_helm_charts(&version.replace("+", "-"))?;
+            }
```

Comment:
@garypen / others: Does anyone know if this was introduced with a arch/os/env check because "When someone tries to build a nightly image using our `xtask` tooling on their local macOS laptop we have no desire to worry them with Helm?" or **because** we wanted to gate it to "the part of the process that runs nightly builds in our CI"?

Reply:
git can remember things that I can't: https://github.com/apollographql/router/pull/3313/commits/81765d5d6dd9e80f86ad4bff8ecd44efe7a63cc1

The main motivation is thus to avoid CI failures (on Windows) during nightly builds.

Reply:
I had looked at that.  What about Windows + Helm is incompatible?

Reply:
I don't think anything is incompatible, we just don't want to do it. We are only producing a single helm chart, and we choose to produce it from the linux/amd_64 branch of the CI build.

I think the reason it fails in CI is because we don't install helm as part of our CI builds apart from on linux/amd_64.

Reply:
Got it.  That last detail is the details of the **why** that I was looking for that weren't captured within the commit/code which makes the entire existence of this gate make sense to someone visiting this for the first time.